internal/bminventory/inventory.go

```diff
@@ -197,6 +230,15 @@ func (b *bareMetalInventory) formatIgnitionFile(cluster *common.Cluster, params
 		"NoProxy":              cluster.NoProxy,
 		"SkipCertVerification": strconv.FormatBool(b.SkipCertVerification),
 	}
+
+	if b.Config.DeployTarget == "onprem" {
```

Comment:
This no longer exists, the idea is to keep this file agnostic as to where it's deployed.  I suggest adding to the Config struct an option called, say ServiceIPs, which is by default empty.  Then you can add /etc/hosts if the option was set.

Reply:
I think that we should avoid using DeployTarget in the code.
It will be cleaner to have an interface that you init in the main. 
And  why is it IPs? It's more then one?

Reply:
Yes, there will be multiple IPs ( all IPV4 and IPV6) added in the /etc/hosts which will resolve the custom dns name `assisted-api.local.openshift.io`